test/distributed/fsdp/test_fsdp_mixed_precision.py

```diff
@@ -927,6 +927,97 @@ def test_float16_on_one_submodule_skip_inputs_error(self):
         ):
             fsdp(x).sum().backward()
 
+    @skip_if_lt_x_gpu(2)
+    def test_submodules_with_different_precisions_error(self):
+        forward_inputs: Dict[nn.Module, torch.Tensor] = {}
+        float16 = MixedPrecision(param_dtype=torch.float16, cast_forward_inputs=True)
+        float8 = MixedPrecision(param_dtype=torch.float8, cast_forward_inputs=True)
+
+        model = SaveForwardInputsModel(
+            forward_inputs=forward_inputs, cast_forward_inputs=False
+        ).cuda()
+        c1, c2 = model.c1, model.c2
+        x = torch.zeros(2, 100, device="cuda")
+
+        # For submodules with different precisions, right now current design
+        # does not support the case when the root FSDP instance wraps a submodule
+        # that is not the first one executed. Because for that submodule, its inputs
+        # (or previous submodule's outputs) have no way to be casted, instead,
+        # the root module's inputs are casted upfront before entering
+        # root module's forward
+        model.c1 = FSDP(model.c1, mixed_precision=float16)
+        fsdp = FSDP(model, mixed_precision=float8)
+        with self.assertRaisesRegex(
+            RuntimeError, "mat1 and mat2 must have the same dtype"
+        ):
+            fsdp(x).sum().backward()
+
+    @skip_if_lt_x_gpu(2)
+    def test_submodules_with_different_precisions(self):
+        forward_inputs: Dict[nn.Module, torch.Tensor] = {}
+        float16 = MixedPrecision(param_dtype=torch.float16, cast_forward_inputs=True)
+        float8 = MixedPrecision(param_dtype=torch.float8, cast_forward_inputs=True)
```

Comment:
Do we have any concern for `float8` not being supported for some versions of CI?

Reply:
To double check, do you mean `float32`? I think `bfloat16` is not supported for some CI as well.